lib/src/components/NxToggle/__tests__/NxToggleRTL.test.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import NxToggle, { Props } from '../NxToggle';
+
+describe('NxToggle', function() {
+  const simpleProps: Props = {
+    inputId: 'toggle-id',
+    isChecked: false,
+    onChange: () => {},
+    disabled: undefined,
+    children: 'Enables whales'
+  };
+
+  it('renders a <label> with an <input>', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
+
+    const checkbox = screen.getByRole<HTMLInputElement>('switch', { name: 'Enables whales' });
+
+    expect(checkbox.tagName).toEqual('INPUT');
+
+    const label = checkbox.labels?.[0];
+
+    expect(label?.textContent).toEqual('Enables whales');
+  });
+
+  it('renders an input element with role switch with the correct attributes and classname', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
+
+    const checkbox = screen.getByRole('switch');
+
+    expect(checkbox).toHaveAttribute('type', 'checkbox');
+    expect(checkbox).toHaveAttribute('id', 'toggle-id');
+    expect(checkbox).toHaveClass('nx-toggle__input');
+  });
+
+  it('adds classes specified with the className prop', function() {
+    const { container } = render(<NxToggle {...simpleProps} className="foo"></NxToggle>);
+
+    const label = container.querySelector('label');
+
+    expect(label).toHaveClass('foo');
+    expect(label).toHaveClass('nx-toggle');
+  });
+
+  it('calls its onChange prop when the input fires a change event', async function() {
+    const onChange = jest.fn();
+
+    render(<NxToggle {...simpleProps} onChange={onChange}></NxToggle>);
+
+    const checkbox = screen.getByRole('switch');
+
+    expect(onChange).not.toHaveBeenCalled();
+
+    await userEvent.click(checkbox);
+
+    expect(onChange).toHaveBeenCalled();
+  });
+
+  it('passes input attributes into the input element and does not clash with top-level attributes', function() {
+    const props = {
+      inputId: 'not-garfield',
+      disabled: true,
+      isChecked: true,
+      className: 'label-classname',
+      inputAttributes: {
+        id: 'garfield',
+        name: 'garfield',
+        disabled: false,
+        className: 'input-classname',
+        checked: false
+      } as Props['inputAttributes']
+    };
```

Comment:
> I haven't tried it but I think this should work and would be better if it does:

Unfortunately I got this :(

<img width="956" alt="Screen Shot 2022-06-22 at 9 57 46 AM" src="https://user-images.githubusercontent.com/10160030/175047456-e3496767-4952-49e8-bffc-2f3c2707db7d.png">

Reply:
> Unfortunately I got this :(

That's because we explicitly don't allow `disabled` to be specified on the `inputAttributes` of `NxToggle`, since it's part of the main props.
```
export type InputAttributesProp =
  Omit<InputHTMLAttributes<HTMLInputElement>,
  'disabled' | 'checked' | 'readonly' | 'onChange'>;
```

Reply:
> > Unfortunately I got this :(
> 
> That's because we explicitly don't allow `disabled` to be specified on the `inputAttributes` of `NxToggle`, since it's part of the main props.
> 
> ```
> export type InputAttributesProp =
>   Omit<InputHTMLAttributes<HTMLInputElement>,
>   'disabled' | 'checked' | 'readonly' | 'onChange'>;
> ```

Updated: ee21ee3bedf888680b74d745104e502641f677ee